### ISSUE TITLE
Preact / Deno Fresh support

### ISSUE THREAD

**Issue by NuroDev** [ascorbic/og-edge#8]
Testing out implementing this project into my own project using [Deno Fresh](https://fresh.deno.dev/) & [Preact](https://preactjs.com/) & getting the following error due to the type mismatch between React & Preact components:
```ts
Argument of type 'Element' is not assignable to parameter of type 'ReactElement<any, string | JSXElementConstructor<any>>'.
  Types of property 'type' are incompatible.
    Type 'string | ComponentType<any>' is not assignable to type 'string | JSXElementConstructor<any>'.
      Type 'ComponentClass<any, {}>' is not assignable to type 'string | JSXElementConstructor<any>'.
        Type 'ComponentClass<any, {}>' is not assignable to type 'new (props: any) => Component<any, any, any>'.
          Property 'refs' is missing in type 'Component<any, {}>' but required in type 'Component<any, any, any>'.deno-ts(2345)
```

I am assuming this is a bug. I could of course probably just add `react` as an import & cast the type passed in but wanted to report it to see if this something we could possibly improve natively in the module so type casting is not needed.

**Comment by ascorbic** [comment]
Yeah, the problem is that the Satori library [expects a ReactElement](https://github.com/vercel/satori/blob/main/src/satori.ts#L44), so we need to use the same type. Casting should be fine though. Import the `ReactElement` type and cast it to that. Alternatively, use react as the jsx Import source just for the og image file.
